### DISCUSSION TITLE
Upload zip or tar.gz file in POST

### DISCUSSION THREAD

**Discussion by guzman109** [litestar-org/litestar#3250] [Q&A] [answered]
I have the following to be able to upload a .tar.gz file.
`@post("/upload", media_type="application/x-gtar")
    async def upload(self,
        data: Annotated[UploadFile, Body(media_type=RequestEncodingType.MULTI_PART)]
    ) -> dict[str, Any]:
        return { 
            "File Name": data.filename,
            "Content Type": data.content_type,
            "Headers": data.headers
        }`
However, I keep getting 500 internal server errors. Am I implementing this correctly?

**Comment by Alc-Alc** [comment, marked as answer]
`media_type` is used to serialize whatever you return from the handler `upload`. Litestar does not know how to serialize `dict[str, Any]` into a tar file which is why it fails.

`media_type` is not used to indicate that your route accepts tar gz files.

You can remove `media_type` and you will notice that you can indeed upload a file and get a json response

**Reply by guzman109** [reply]
OMG Thank you so much!
